extensions/scheduler/common/src/main/java/io/quarkus/scheduler/common/runtime/util/SchedulerUtils.java

```diff
@@ -98,6 +102,21 @@ public static ZoneId parseCronTimeZone(Scheduled scheduled) {
         return timeZone.equals(Scheduled.DEFAULT_TIMEZONE) ? null : ZoneId.of(timeZone);
     }
 
+    public static <T> T instantiateBeanOrClass(Class<T> type) {
+        Instance<T> instance = Arc.container().select(type, Any.Literal.INSTANCE);
+        if (instance.isAmbiguous()) {
+            throw new IllegalArgumentException("Multiple beans match the type: " + type);
+        } else if (instance.isUnsatisfied()) {
+            try {
+                return type.getConstructor().newInstance();
+            } catch (Exception e) {
+                throw new IllegalStateException("Unable to instantiate the class: " + type);
```

Comment:
I will merge this but I wonder if we will regret to not have included the expection as a cause :).

Reply:
We'll see.. :shrug: